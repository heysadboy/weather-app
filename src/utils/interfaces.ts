import { EActionType, EStatusType } from './enums';

export interface IAction {
    type: EActionType | EStatusType,
    payload: IWeather[] | IWeather | IPage | IStatus
}

export interface IWeather {
    id: number,
    temp: number,
    temp_f: number,
    temp_c: number,
    dt_txt: string,
    tm_txt: string
}

export interface IPage {
    start: number,
    end: number
}

export interface IStatus {
    code: EStatusType
    message: string
}