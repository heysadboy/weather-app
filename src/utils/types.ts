import combineReducers from '../reducers';

export type AppState = ReturnType<typeof combineReducers>;