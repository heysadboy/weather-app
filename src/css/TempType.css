#temp-type-container {
    padding: 20px 100px 20px 100px;
}

#f-temp-input {
    font-size: medium;
}

#t-temp-input {
    font-size: medium;
}

@media screen and (max-width: 768px) {
    #temp-type-container {
        padding: 20px 40px 20px 40px;
    }
}