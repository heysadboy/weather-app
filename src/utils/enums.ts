export enum EActionType {
    fetch_weather = 'FETCH_WEATHER',
    temp_data = 'SET_TEMP_DATA',
    left_page = 'LEFT_PAGE',
    right_page = 'RIGHT_PAGE'
}

export enum EStatusType {
    error = 'ERROR',
    loading = 'LOADING',
    ok = 'OK'
}

export enum ETempType {
    f = "Fahrenheit",
    c = "Celsius"
}