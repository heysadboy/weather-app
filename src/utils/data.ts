import { IPage, IStatus } from '../utils/interfaces';
import { EStatusType } from './enums';


export const INITIAL_PAGE: IPage = { start: 0, end: 2 }

export const INITAIL_STATUS: IStatus = { code: EStatusType.loading, message: "Loading"}

export const APP_ID = '75f972b80e26f14fe6c920aa6a85ad57';
