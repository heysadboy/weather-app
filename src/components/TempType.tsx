import { FC } from "react";
import { ETempType } from "../utils/enums";
import '../css/TempType.css';

interface ITempTypeProps {
    tempType: ETempType,
    setTempType: (tempType: ETempType) => void
}

const TempType: FC<ITempTypeProps> = ({ setTempType, tempType }) => {

    const tempTypeController = (e: any) => {
        setTempType(e.target.value);
    };

    return (
        <div id="temp-type-container" className="ui two column grid">
            <div className="row grouped fields" onChange={tempTypeController}>
                <div className="left algined column field">
                    <div id="f-temp-input" className="ui radio checkbox">
                        <input type="radio" value={ETempType.f} name="temperature-choice" defaultChecked={tempType === ETempType.f ? true : false} />
                        <label>{ETempType.f}</label>
                    </div>
                </div>
                <div className="right aligned column field">
                    <div id="t-temp-input" className="ui radio checkbox">
                        <input type="radio" value={ETempType.c} name="temperature-choice" defaultChecked={tempType === ETempType.c ? true : false} />
                        <label>{ETempType.c}</label>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default TempType;