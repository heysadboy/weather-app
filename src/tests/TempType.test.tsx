import renderer from 'react-test-renderer';
import TempType from '../components/TempType';
import { ETempType } from '../utils/enums';


test('Test Temp Type', () => {
    const tempType: ETempType = ETempType.c;
    const setTempType = () => { };

    const component = renderer.create(
        <TempType tempType={tempType} setTempType={setTempType} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});