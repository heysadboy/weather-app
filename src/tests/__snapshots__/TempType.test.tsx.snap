// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Temp Type 1`] = `
<div
  className="ui two column grid"
  id="temp-type-container"
>
  <div
    className="row grouped fields"
    onChange={[Function]}
  >
    <div
      className="left algined column field"
    >
      <div
        className="ui radio checkbox"
        id="f-temp-input"
      >
        <input
          defaultChecked={false}
          name="temperature-choice"
          type="radio"
          value="Fahrenheit"
        />
        <label>
          Fahrenheit
        </label>
      </div>
    </div>
    <div
      className="right aligned column field"
    >
      <div
        className="ui radio checkbox"
        id="t-temp-input"
      >
        <input
          defaultChecked={true}
          name="temperature-choice"
          type="radio"
          value="Celsius"
        />
        <label>
          Celsius
        </label>
      </div>
    </div>
  </div>
</div>
`;
